src/tensor.c

```diff
@@ -924,7 +924,7 @@ int RAI_TensorReplyWithValues(RedisModuleCtx *ctx, RAI_Tensor *t) {
         double val;
         for (i = 0; i < len; i++) {
             int ret = RAI_TensorGetValueAsDouble(t, i, &val);
-            if (!ret) {
+            if (ret == 1) {
```

Comment:
I found a mistake in RAI_TensorGetValueAsDouble. In the documentation it says that it returns 0 in case of success, but in practice it was implemented such that it returns 1 in case of success. So I changed it to be as it suppose to be according to the documentation, and therefore this change was needed as well.